docs/source/configuration/external_timelock_service_configs/timelock_server_config.rst

```diff
@@ -131,6 +131,46 @@ default values.
      - The length of time between a follower initiating a ping to a leader and, if it hasn't received a response,
        believing the leader is down, in ms (default: ``5000``).
 
+.. _timelock-server-time-limiting:
+
+Time Limiting
+-------------
+
+Clients that make long-running lock requests will block a thread on TimeLock for the duration of their request. More
+significantly, if these requests are blocked for longer than the idle timeout of the server's application connector
+on HTTP/2, then Jetty will send a stream closed message to the client. This can lead to an infinite buildup of threads
+and was the root cause of issue `#1680 <https://github.com/palantir/atlasdb/issues/1680>`__. We thus reap the thread
+for interruptible requests before the timeout expires, and send an exception to the client indicating that its request
+has timed out, but it is free to retry on the same node. Note that this issue may still occur if a *non-interruptible*
```

Comment:
Yep, I _think_ the only interruption point should be when blocked on the lock-sync.